Sources/macro.js

```diff
@@ -722,30 +733,37 @@ export function event(publicAPI, model, eventName) {
     }
     /* eslint-disable prefer-rest-params */
     for (let index = 0; index < callbacks.length; ++index) {
-      const cb = callbacks[index];
-      if (cb) {
-        cb.apply(publicAPI, arguments);
+      const [, cb, priority] = callbacks[index];
+      if (priority < 0) {
+        setTimeout(() => cb.apply(publicAPI, arguments), 1 - priority);
+      } else if (cb) {
+        // Abort only if the callback explicitly returns false
+        const continueNext = cb.apply(publicAPI, arguments);
+        if (continueNext === EVENT_ABORT) {
+          break;
+        }
       }
     }
     /* eslint-enable prefer-rest-params */
   }
 
   publicAPI[`invoke${capitalize(eventName)}`] = invoke;
 
-  publicAPI[`on${capitalize(eventName)}`] = (callback) => {
+  publicAPI[`on${capitalize(eventName)}`] = (callback, priority = 0.0) => {
     if (model.deleted) {
       vtkErrorMacro('instance deleted - cannot call any method');
       return null;
     }
 
-    const index = callbacks.length;
-    callbacks.push(callback);
-    return on(index);
+    const callbackID = curCallbackID++;
+    callbacks.push([callbackID, callback, priority]);
+    callbacks.sort(([, cb1, pri1], [, cb2, pri2]) => pri2 - pri1);
```

Comment:
your sort does not use cb1 and cb2...

Reply:
what about doing it like that

```
callbacks.sort((a, b) => b[2] - a[2]);
```

With maybe the function be only defined once so it does not get created at each call?

Reply:
The comparator is small enough do not need a separate function. Will there be a measurable performance hit? Maybe if on* gets called a lot...

Reply:
I agree, and I'm not expecting perf hit.

Reply:
:+1: